File: src/tools/organizations/setActive.ts
import type { z } from "zod";
import type { Context, Tool } from "../types";
import { OrganizationSetActiveSchema } from "../../schema/tool-inputs";

const schema = OrganizationSetActiveSchema;

type Params = z.infer<typeof schema>;

export const setActiveHandler = async (context: Context, params: Params) => {
	const { orgId } = params;
	await context.cache.set("orgId", orgId);

	return {
		content: [{ type: "text", text: `Switched to organization ${orgId}` }],
	};
};

const tool = (): Tool<typeof schema> => ({
	name: "organization-set-active",
	description: `
        - Use this tool to set the active organization.
    `,
	schema,
	handler: setActiveHandler,
});

export default tool;

File: python/tools/organizations/set_active.py
from schema.tool_inputs import OrganizationSetActiveSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def set_active_handler(context: Context, params: OrganizationSetActiveSchema) -> ToolResult:
    org_id = str(params.orgId)

    await context.cache.set("org_id", org_id)

    return ToolResult(content=f"Switched to organization {org_id}")


def set_active_org_tool() -> Tool[OrganizationSetActiveSchema]:
    definition = get_tool_definition("organization-set-active")

    return Tool(
        name="organization-set-active",
        description=definition["description"],
        schema=OrganizationSetActiveSchema,
        handler=set_active_handler,
    )
